### PR TITLE
docs: remove references to SettingsType

### PR DESCRIPTION
**Pull request by rgraber**
The SettingsType enum is not present in edx-django-utils. Remove references to it and explain that the values are available in edx-platform. Also explain that other IDAs may require other values.

No version upgrade since this is just docs.

Issue: https://github.com/openedx/edx-django-utils/issues/288

### CODE REVIEW

File: edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst [hunk 1]
```diff
@@ -32,12 +32,13 @@ file::
...
 
-4. configure the Plugin App in their AppConfig
+4. configure the Plugin App in their AppConfig. Note that in this example, we are explicitly configuring plugins for use in edx-platform. If your plugin is going to be used in another IDA, you may have different project and settings types. You will need to look at the IDA in question for what values it expects. You may want to add new values to the relevant enums.
+
 class::
 
    from django.apps import AppConfig
    from edx_django_utils.plugins.constants import (
-       ProjectType, SettingsType, PluginURLs, PluginSettings, PluginContexts
+       ProjectType, PluginURLs, PluginSettings, PluginContexts
```
Comment (robrap) [inline edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: Actually, if we are going to leave this as an LMS example, why not leave `ProjectType` and `SettingsType` as-is, except moving them to a separate import that is edx-platform specific. You might even move your note to be a code comment against the platform specific import statement. Thoughts?

Comment (rgraber) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: I'm not sure we want to encourage importing from edx-platform

Comment (robrap) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: Yes. I forgot this is plugin code. I was confused by `ProjectType` being left as-is. Should that also be converted to strings?

Comment (robrap) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: Also, I wonder if the ultimate solution is to just move all the constants to the library? What is the down side of that? Not sure if we'd need LMSSettingsType, etc. Anyway, I know that affects scope of this, but it is also difficult to use the example code if we leave a bunch of strings and it isn't clear where they are coming from.

Comment (rgraber) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: The downside that I can see is the precedent of having IDA-specific code in plugins that should theoretically be able to go in any IDA. I don't think we'd want to add constants every time we enabled plugins in a new IDA that had its own configuration vars.

Comment (rgraber) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: For now I'd just as soon remove ProjectType, although it actually appears in a few more places in the documentation, so I should remove it there too.

Comment (robrap) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: Sure. Maybe just add a code comment next to the first of the strings pointing to `ProjectType` so someone knows where they are coming from?

File: edx_django_utils/plugins/plugin_contexts.py [hunk 2]
```diff
@@ -19,8 +19,10 @@ def get_plugins_view_context(project_type, view_name, existing_context=None):
     selected function to get their context dicts.
 
     Params:
-        project_type: a string that determines which project (lms or studio) the view is being called in. See the
-            ProjectType enum in plugins/constants.py for valid options
+        project_type: a string that determines which project the view is being called in. See the
+            ProjectType enum in
+            https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/plugins/constants.py#L14 for
```
Comment (robrap) [inline edx_django_utils/plugins/plugin_contexts.py:24]: 1. Also make this a permalink.
2. It is strange that there isn't an equivalent doc needed somewhere for the SettingsType.

File: edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst (hunk 1 above)
Comment (robrap) [reply edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:41]: Doesn't `ProjectType` also need to be dropped from the import?

File: edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst [hunk 3]
```diff
@@ -71,17 +74,19 @@ class::
            # Configuration setting for Plugin Settings for this app.
            PluginSettings.CONFIG: {
 
-               # Configure the Plugin Settings for each Project Type, as needed.
-               ProjectType.LMS: {
+               # Configure the Plugin Settings for each Project Type, as needed. The full list setting types for edx-platform is
```
Comment (robrap) [inline edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:77]:
```suggestion
               # Configure the Plugin Settings for each Project Type, as needed. The full list of setting types for edx-platform is 
```

File: edx_django_utils/plugins/plugin_contexts.py [hunk 4]
```diff
@@ -19,8 +19,11 @@ def get_plugins_view_context(project_type, view_name, existing_context=None):
     selected function to get their context dicts.
 
     Params:
-        project_type: a string that determines which project (lms or studio) the view is being called in. See the
-            ProjectType enum in plugins/constants.py for valid options
+        project_type: a string that determines which project the view is being called in. See the
+            ProjectType enum in
+            https://github.com/openedx/edx-platform/blob/2dc79bcab42dafed2c122eb808cdd5604327c890/openedx/core/
+            djangoapps/plugins/constants.py#L14 .
+            for valid options for edx-platform
```
Comment (robrap) [inline edx_django_utils/plugins/plugin_contexts.py:26]:
```suggestion
            djangoapps/plugins/constants.py#L14
            for valid options for edx-platform.
```

File: edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst [hunk 5]
```diff
@@ -71,17 +75,20 @@ class::
            # Configuration setting for Plugin Settings for this app.
            PluginSettings.CONFIG: {
 
-               # Configure the Plugin Settings for each Project Type, as needed.
-               ProjectType.LMS: {
+               # Configure the Plugin Settings for each Project Type, as needed. The full list of setting types for edx-platform is
+               # here:
+               # https://github.com/openedx/edx-platform/blob/2dc79bcab42dafed2c122eb808cdd5604327c890/openedx/core/djangoapps/plugins/constants.py#L25
```
Comment (robrap) [inline edx_django_utils/plugins/docs/how_tos/how_to_create_a_plugin_app.rst:80]: Consider adding a period, as above.

**Review by robrap — APPROVED** [review]
Minor issues with periods, but looks good. Thank you!

**Pull request merged**